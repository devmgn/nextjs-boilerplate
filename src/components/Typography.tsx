import styled from '@emotion/styled';
import createShouldForwardProp from '@/utils/createShouldForwardProp';
import type { TupleToUnion } from 'type-fest';

const VARIANTS = ['title', 'body', 'caption'] as const;

type TypographyProps = {
  variant?: TupleToUnion<typeof VARIANTS>;
};

const Typography = styled('p', {
  shouldForwardProp: createShouldForwardProp('variant'),
})<TypographyProps>`
  font-size: ${({ variant }) => {
    if (variant === 'title') {
      return '24px';
    }
    if (variant === 'caption') {
      return '10px';
    }
    return '14px';
  }};
  line-height: 1.5;
`;

export default Typography;
