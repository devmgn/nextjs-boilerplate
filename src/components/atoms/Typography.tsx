import styled from '@emotion/styled';

const Typography = styled('p')`
  font-size: 16px;
  line-height: 1.5;
`;

export default Typography;
