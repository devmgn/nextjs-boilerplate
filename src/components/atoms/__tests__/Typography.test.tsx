import { render } from '@testing-library/react';
import Typography from '../Typography';

describe('Typography', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<Typography />);
    expect(asFragment()).toMatchSnapshot();
  });
});
