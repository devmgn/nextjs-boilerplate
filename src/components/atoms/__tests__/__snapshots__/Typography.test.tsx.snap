// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography スナップショットテスト 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 16px;
  line-height: 1.5;
}

<p
    class="emotion-0 emotion-1"
  />
</DocumentFragment>
`;
